### PR TITLE
Rename default branch to `main`

### PR DESCRIPTION
**Pull request by malfet**
Mostly `s/@master/@main` in numerous `.yml` files

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
CI would probably work, but a bunch of queries on Rockset definitely need to be updated eventually.  HUD could be broken otherwise

File: .circleci/README.md [hunk 1]
```diff
@@ -308,7 +308,7 @@ Note that the Windows Python wheels are still built in conda environments. Some
 
 * These should all be consolidated
 * These must run on all OS types: MacOS, Linux, and Windows
-* These all run smoke tests at the moment. They inspect the packages some, maybe run a few import statements. They DO NOT run the python tests nor the cpp tests. The idea is that python tests on master and PR merges will catch all breakages. All these tests have to do is make sure the special binary machinery didn’t mess anything up.
+* These all run smoke tests at the moment. They inspect the packages some, maybe run a few import statements. They DO NOT run the python tests nor the cpp tests. The idea is that python tests on mainand PR merges will catch all breakages. All these tests have to do is make sure the special binary machinery didn’t mess anything up.
```
Comment (PaliC) [inline .circleci/README.md:311]: typo

File: .circleci/README.md [hunk 2]
```diff
@@ -340,7 +340,7 @@ The Dockerfiles are available in pytorch/builder, but there is no circleci job o
...
+Sometimes we want to push a change to mainand then rebuild all of today's binaries after that change. As of May 30, 2019 there isn't a way to manually run a workflow in the UI. You can manually re-run a workflow, but it will use the exact same git commits as the first run and will not include any changes. So we have to make a PR and then force circleci to run the binary workflow instead of the normal tests. The above PR is an example of how to do this; essentially you copy-paste the binarybuilds workflow steps into the default workflow steps. If you need to point the builder repo to a different commit then you'd need to change https://github.com/pytorch/pytorch/blob/main/.circleci/scripts/binary_checkout.sh#L42-L45 to checkout what you want.
```
Comment (PaliC) [inline .circleci/README.md:343]: typo

**Review by PaliC — APPROVED** [review]

**Comment by justinchuby** [comment]
Need to change https://github.com/pytorch/pytorch/blob/efc3887ea508b3cfd94603fd8afe4e8cf6dce7b7/.lintrunner.toml#L1 as well?

**Pull request merged**